src/System.Buffers/tests/ArrayPool/CollectionTests.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Linq;
+using Xunit;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace System.Buffers.ArrayPool.Tests
+{
+    [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+    public class CollectionTests : RemoteExecutorTestBase
+    {
+        private const string TrimSwitchName = "DOTNET_SYSTEM_BUFFERS_ARRAYPOOL_TRIMSHARED";
+
+        [Theory,
+            InlineData(true),
+            InlineData(false)]
```

Comment:
OuterLoop for all of these.  I'm hesitant to have tests that take so long even in OuterLoop, but I know I differ there from @danmosemsft.

Reply:
This one I will for sure, I called out in the comments that I'm leaving it innerloop until I see it go CI green. The other one is actually fast (a few seconds).